adaptors task/student/src/impl.cpp

```diff
@@ -6,30 +6,31 @@ const std::vector<std::pair<char, char>> brackets {
     {'[', ']'},
     {'{', '}'}
 };
+
 bool isValid(const std::string& source){
-    if (source.empty() or source.size()==1)
+    if (source.size()==1)
     {
         return false;
     }
-    std::stack<char> res;
+    std::stack<char> checkString;
 
-    for (auto stringCh : source) {
-        for (auto reg : brackets) {
-            if (stringCh == reg.first)
-                {
-                    res.push(stringCh);
-                    continue;
-                }
-                else if (stringCh == reg.second)
+    for (auto currentChar : source) {
+        for (auto bracket : brackets) {
+            if (currentChar == bracket.first)
+            {
+                checkString.push(currentChar);
+                continue;
+            }
+            else if (currentChar == bracket.second)
+            {
+                if (source.at(0) == bracket.second or checkString.top() != bracket.first)
```

Comment:
why source.at(0)???

Reply:
I changed it to checkString.empty() in the last commit